### PR TITLE
Add percentages to support PAC snapshot

### LINKED ISSUE #4877: Make API changes to support high-level PAC data
Make API changes to support high-level PAC data

/committee/<committe_id>/totals (example: https://api.open.fec.gov/v1/committee/C00344234/totals/?api_key=DEMO_KEY&per_page=100&page=1)

**Receipts**

- `individual_contributions_percent`:  Line 11(a)(iii)  divided by Line 19
-  `party_and_other_committee_contributions_percent`: (Line 11(b) + Line 11(c)) divided by Line 19

**Disbursements**

- `contributions_ie_and_party_expenditures_made_percent`: Line 23 + 24 + 25 divided by Line 31
- `operating_expenditures_percent`:  Line 21(c)  divided by Line 31

Endpoint: `/committee/<string:committee_id>/totals/`
View: `totals.TotalsCommitteeView`
Model: `models.CommitteeTotalsPacParty`
DB View: `ofec_totals_pac_party_vw`

### PR DESCRIPTION
**Pull request by lbeaufort**
## Summary (required)

Resolves #4877

- Add percentages to support PAC snapshot
- Handle unusual data scenarios (see #4876)

### Required reviewers

Two developers requested, others welcome

## Impacted areas of the application

General components of the application that this PR will affect:

Add 4 new fields to /committee/<committee_id/totals endpoint (feeds committee profile page totals):
- `individual_contributions_percent`
- `party_and_other_committee_contributions_percent`             
- `contributions_ie_and_party_expenditures_made_percent`                
- `operating_expenditures_percent`

## How to test

- Check out this branch
- Test unusual data scenarios: https://docs.google.com/spreadsheets/d/18Wt98l1lN4tMCAzzLcQOMpCxpAT5I_zoLGZAfiKPVc4/edit#gid=0
- Test a few committees: http://localhost:5000/v1/committee/<EXAMPLE_COMMITTEE_ID>/totals/?cycle=<EXAMPLE>

### CODE REVIEW

File: webservices/common/models/totals.py [hunk 1]
```diff
@@ -216,6 +216,44 @@ class CommitteeTotalsPacParty(CommitteeTotals):
...
+    @property
+    def contributions_ie_and_party_expenditures_made_percent(self):
+        """  (Line 23 + 24 + 25) divided by Line 31 """
+
+        numerators = [
+            self.fed_candidate_committee_contributions,
+            self.independent_expenditures,
+            self.coordinated_expenditures_by_party_committee,
+        ]
+        denominators = [self.disbursements,]
```
Comment (lbeaufort) [inline webservices/common/models/totals.py:246]: take out comma

File: tests/test_utils.py [hunk 2]
```diff
@@ -323,3 +325,38 @@ def test_env_var_split(self):
...
+            # Null values
+            ([None], [None], None),
+            ([0], [None], None),
+            ([None], [100], None),
+            ([], [], None),
+            ([1, None, 3], [10], None),
+        ]
+        for test_case in test_cases:
+            numerator, denominator, expected = test_case
+            result = utils.get_percentage(numerator, denominator)
```
Comment (lbeaufort) [inline tests/test_utils.py:357]: `get_percentage(numerators, denominators)`

File: webservices/utils.py [hunk 3]
```diff
@@ -545,3 +545,33 @@ class DateTimeEncoder(JSONEncoder):
     def default(self, obj):
         if isinstance(obj, (datetime.date, datetime.datetime)):
             return obj.isoformat()
+
+
+def get_percentage(numerators, denominators):
+    """Calculate the percentage numerator (top) is of denominator (bottom), 2 decimals
```
Comment (lbeaufort) [inline webservices/utils.py:551]: `numerators (top) are of denominators (bottom)`

File: webservices/utils.py [hunk 4]
```diff
@@ -545,3 +545,33 @@ class DateTimeEncoder(JSONEncoder):
...
+
+    For unusual data scenarios, return None
```
Comment (lbeaufort) [inline webservices/utils.py:553]: `If unable to calculate, return None`

File: webservices/utils.py [hunk 5]
```diff
@@ -545,3 +545,33 @@ class DateTimeEncoder(JSONEncoder):
...
+    """
+
+    # Values must be in a list
+    if not isinstance(numerators, list) and denominators(numerators, list):
```
Comment (lbeaufort) [inline webservices/utils.py:557]: `isinstance(denominators, list):`

Make sure this works!

**Review by fec-jli — APPROVED** [review]
work as expected. Great work.Thanks

**Review by hcaofec — APPROVED** [review]
Tested this PR as instructed.  Looks good!

**Comment by lbeaufort** [comment]
Approved, so merging.

**Pull request merged**